api/vehicles.go

```diff
@@ -115,3 +115,36 @@ func (api *API) UpdatesHandler(w http.ResponseWriter, r *http.Request) {
 	// Convert updates to JSON
 	WriteJSON(w, updates) // it's good to take some REST in our server :)
 }
+
+// HistoryHandler updates for last 30 days to endpoint
+func (api *API) HistoryHandler(w http.ResponseWriter, r *http.Request){
+	vehicles, err := api.ms.EnabledVehicles()
+	if err != nil {
+		log.WithError(err).Error("Unable to get enabled vehicles")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	history := make([][]*shuttletracker.Location, 0, len(vehicles))
+	for _, vehicle := range vehicles {
+		since := time.Now().Add(time.Minute * -43200)
+		vehicleUpdates, err := api.ms.LocationsSince(vehicle.ID, since)
+		if err != nil{
+			log.WithError(err).Error("Unable to get last vehicle update.")
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		if len(vehicleUpdates) > 0 {
+			history = append(history, vehicleUpdates)
+		}
+	}
+
+	writerErr := WriteJSON(w, history)
```

Comment:
Minor, but I think changing this to just reuse the same err variable makes this more consistent with the rest of the code base.

Reply:
Will do !